crates/uv/src/commands/build.rs

```diff
@@ -195,11 +224,101 @@ async fn build_impl(
             .root()
             .clone();
 
-        Source::Directory(Cow::Owned(project))
+        vec![Source::Directory(Cow::Owned(project))]
+    } else if all {
+        if matches!(src, Source::File(_)) {
+            return Err(anyhow::anyhow!(
+                "Cannot specify a `--all` when building from a file"
+            ));
+        }
+        let workspace = match workspace {
+            Ok(ref workspace) => workspace,
+            Err(err) => {
+                return Err(
+                    anyhow::anyhow!("`--all` was provided, but no workspace was found")
+                        .context(err),
+                )
+            }
+        };
+        workspace
+            .packages()
+            .values()
+            .map(|package| Source::Directory(Cow::Owned(package.root().clone())))
+            .collect()
     } else {
-        src
+        vec![src]
     };
 
+    let build_futures = packages.iter().map(|src| {
+        build_package(
+            src,
+            output_dir,
+            python_request,
+            no_config,
+            workspace.as_ref(),
+            python_preference,
+            python_downloads,
+            cache,
+            printer,
+            index_locations,
+            &client_builder,
+            hash_checking,
+            build_constraints,
+            no_build_isolation,
+            no_build_isolation_package,
+            native_tls,
+            connectivity,
+            index_strategy,
+            keyring_provider,
+            allow_insecure_host,
+            exclude_newer,
+            sources,
+            concurrency,
+            build_options,
+            sdist,
+            wheel,
+            dependency_metadata,
+            link_mode,
+            config_setting,
+        )
+    });
+
+    let results = futures::future::join_all(build_futures).await;
```

Comment:
You could also sort the results for determinism here right? Generally I'm pretty into having deterministic output if it's not high-cost.

Reply:
I think the build output itself would be interleaved even if sorted. We'd need `--no-build-output` from your prior PR... then we could make the _asset names_ deterministic, I think.

Reply:
We would need that.. wouldn't we :p I feel like in this case it'd probably make sense to do something else with the build output by default instead of interleaving it... like show progress bars for each build and write the output to a file per project or something. Anyway, that doesn't need to be in scope here.

Cross-linking for reference #7675